.github/workflows/syntax_check_cloud.yml

```diff
@@ -0,0 +1,34 @@
+# This workflow will check for exceptions to raise properly and do not generate syntax errors.
+# Also old code style using parenthesis in class definitions will be checked
+
+name: Syntax checking
+
+on:
+  [push, pull_request]
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, ubuntu-22.04]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Check exceptions
+      # Not related to tests, but to QA in general: Exceptions usually take only 1 argument
+      # So a comma is probably a sign of syntax error and should be replaced by a %
+      run: |
+        grep -IrE --colour 'raise .+".*%.*",' --include=*.py ./src/odemis ./scripts/ ./plugins/
+        if [[ $? -eq 0 ]]; then
+           exit 1
+        fi
+
+    - name: Check old style classes
+      # Only use parenthesis in class definition if there is inheritance (code-style)
```

Comment:
Can you add the following line between `- name:` and `- run:`? This will ensure this check runs, even when the previous check fails.
```
if: ${{ !cancelled() }}
```